packages/ra-ui-materialui/src/input/AutocompleteArrayInput.spec.tsx

```diff
@@ -0,0 +1,619 @@
+                            {...defaultProps}
+                            choices={[
+                                { id: 't', name: 'Technical' },
+                                { id: 'p', name: 'Programming' },
+                            ]}
+                            suggestionComponent={React.forwardRef(
+                                (
+                                    {
+                                        suggestion,
+                                        query,
+                                        isHighlighted,
+                                        ...props
+                                    },
+                                    ref
+                                ) => (
+                                    <div
+                                        {...props}
+                                        ref={ref}
+                                        aria-label={suggestion.name}
+                                    />
+                                )
+                            )}
+                        />
+                    )}
+                />
+            );
+            fireEvent.focus(getByLabelText('resources.posts.fields.tags'));
+            expect(getByLabelText('Technical')).toBeDefined();
+            expect(getByLabelText('Programming')).toBeDefined();
+        });
+    });
+
+    it('should display helperText', () => {
+        const { getByText } = render(
+            <Form
+                onSubmit={jest.fn()}
+                render={() => (
+                    <AutocompleteArrayInput
+                        {...defaultProps}
+                        helperText="Can I help you?"
+                    />
+                )}
+            />
+        );
+        expect(getByText('Can I help you?')).toBeDefined();
+    });
+
+    describe('error message', () => {
+        it('should not be displayed if field is pristine', () => {
+            const { queryByText } = render(
+                <Form
+                    onSubmit={jest.fn()}
+                    render={() => (
+                        <AutocompleteArrayInput
+                            {...defaultProps}
+                            meta={{ touched: false, error: 'Required' }}
+                        />
+                    )}
+                />
+            );
+            expect(queryByText('Required')).toBeNull();
+        });
+
+        it('should be displayed if field has been touched and is invalid', () => {
+            const { queryByText } = render(
+                <Form
+                    onSubmit={jest.fn()}
+                    render={() => (
+                        <AutocompleteArrayInput
+                            {...defaultProps}
+                            meta={{ touched: true, error: 'Required' }}
+                        />
+                    )}
+                />
+            );
+            expect(queryByText('Required')).toBeDefined();
```

Comment:
I'm not at ease with `toBeDefined` testing the opposite of `toBeNull`. I don't want to look for low level JS semantics, so I suggest you use `not.ToBeNull` instead